src/mixins/navigator/tagsProvider.js

```diff
@@ -45,53 +43,57 @@ export default {
       if (hideAvailableTags || selectedTags.length) return [];
       return availableTags;
     },
-    translatableTags: ({ availableTags }) => [HIDE_DEPRECATED, ...availableTags],
+    translatableTags: ({ availableTags }) => [FILTER_TAGS.hideDeprecated, ...availableTags],
   },
   methods: {
     extractTags(nodes, apiChanges) {
-      // extract tags from child nodes
-      const apiChangesTypesSet = new Set(Object.values(apiChanges));
-      const tagsSet = new Set(Object.values(FILTER_TAGS));
-      const generalTags = new Set([HIDE_DEPRECATED]);
-      // when API changes are available, remove the `HIDE_DEPRECATED` option
-      if (apiChangesTypesSet.size) {
-        generalTags.delete(HIDE_DEPRECATED);
-      }
-
+      const possibleTags = new Set(Object.values(FILTER_TAGS));
+      // create categories to order the availableTags
       const availableTags = {
         type: [],
         changes: [],
         other: [],
       };
+
+      // when API changes are available, remove the hide deprecated option
+      if (apiChanges && Object.values(apiChanges).length) {
+        possibleTags.delete(FILTER_TAGS.hideDeprecated);
+      }
+
       // iterate over the nodes to render
       for (const childID in nodes) {
         if (!Object.hasOwnProperty.call(nodes, childID)) {
           // eslint-disable-next-line no-continue
           continue;
         }
         // if there are no more tags to iterate over, end early
-        if (!tagsSet.size && !apiChangesTypesSet.size && !generalTags.size) {
+        if (!possibleTags.size) {
           break;
         }
         // extract props
         const { type, path, deprecated } = nodes[childID];
-        // grab the tagLabel
+
+        // add type tag
         const tag = TOPIC_TYPE_TO_TAG[type];
-        const changeType = apiChanges[path];
-        // try to match a tag
-        if (tagsSet.has(tag)) {
+        if (tag && possibleTags.has(tag)) {
           // if we have a match, store it
           availableTags.type.push(tag);
           // remove the match, so we can end the filter early
-          tagsSet.delete(tag);
+          possibleTags.delete(tag);
         }
-        if (changeType && apiChangesTypesSet.has(changeType)) {
-          availableTags.changes.push(ChangeNames[changeType]);
-          apiChangesTypesSet.delete(changeType);
+
+        // add change tag
+        const changeType = apiChanges[path];
+        if (changeType && FILTER_TAGS[changeType]) {
```

Comment:
```suggestion
        if (changeType && possibleTags.has(changeType)) {
```

Reply:
I didn't check in `possibleTags`, because like you mentioned in the other comment, `possibleTags` is derived from `FILTER_TAGS`. So I just checked if this tag is a valid tag in general. But this reminded me that maybe I can optimize this a bit more.

Reply:
https://github.com/hqhhuang/swift-docc-render/commit/b48a87c9bffdaffd44886902299a36ff53e051cc
Let me know if this makes any sense

Reply:
Actually, I was wrong when I commented the other day, sorry.

I meant instead:
```suggestion
        if (changeType && possibleTags.has(FILTER_TAGS[changeType])) {
```

We need to check if possibleTags continues having the change tag, because if we don't, we will add the same tag duplicated every single time. To avoid that, we delete the tag from possibleTags (our reference Array) but if we don't check again, we will continue adding it.

Since you are repeating `FILTER_TAGS[changeType]` a lot, we could set it up in a `changeType` const.

Reply:
About https://github.com/hqhhuang/swift-docc-render/commit/b48a87c9bffdaffd44886902299a36ff53e051cc, it seems good to me!

Reply:
I committed the suggestion. It was an oversight on my part. Could you give the whole thing another review?